hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -213,11 +214,27 @@ class DefaultSource extends RelationProvider
         classOf[HoodieROTablePathFilter],
         classOf[org.apache.hadoop.fs.PathFilter])
 
+      val specifySchema = if (schema == null) {
+        // Load the schema from the commit meta data.
+        // Here we should specify the schema to the latest commit schema since
+        // the table schema evolution.
+        val schemaUtil = new TableSchemaResolver(metaClient)
+        try {
+          Some(SchemaConverters.toSqlType(schemaUtil.getTableAvroSchema)
+            .dataType.asInstanceOf[StructType])
+        } catch {
+          case _: Throwable =>
+            None // If there is no commit in the table, we can not get the schema
+                 // with schemaUtil, return None here.
+        }
+      } else {
+        None
```

Comment:
shouldn't this be Option.of(schema) ?

Reply:
basically can we have a test which hits this else block(231). I assume all tests hit line 223. schema evol tests hits 227.

Reply:
The schema is null at this if-branch, so it should be None.

Reply:
Thanks for remind me this.